### DISCUSSION TITLE
"appending" dvc.log.lock file type idea for exps

### DISCUSSION THREAD

**Discussion by majidaldo** [treeverse/dvc#10613] [General]
I've been using dvc for a while and I understand the architecture and philosophy. I also understand the decisions behind `dvc exp`eriments with the challenge there being trying to align git's linear development notion with experiment's more "parallel" notion.

I think having to go through dvc exp is too "heavy". Now, if you think of using special git refs as "backend" storage for experiments, alternatively you can achieve the same with dvc lock files that append (instead of overwrite) runs.  This way you can keep original (simple) dvc behavior of just dealing with lock files while also capturing the same information for parameterized runs (exps).

So, I should be able to "checkout" a particular run from the dvc.log.lock file instead of having to deal with dvc exp (parameterized or not).

**Comment by shcheklein** [comment]
Interesting idea @majidaldo . How about changes in code and `.dvc` files though? We need to capture those as well, right?

**Reply by majidaldo** [reply]
.dvc files and dvc.lock files can represent "checkedout" runs.  (but you can have a "logging" version of .dvc files as well)

you can associate code versions on a post-commit hook. (so now you can actually associate a git msg with a run instead of the dvc exp way).
